app/livechat/server/hooks/saveVisitorInactivity.js

```diff
@@ -0,0 +1,25 @@
+import moment from 'moment';
+
+import { callbacks } from '../../../callbacks/server';
+import { LivechatRooms, LivechatDepartment } from '../../../models/server';
+import { settings } from '../../../settings/server';
+
+callbacks.add('afterSaveMessage', function(message, room) {
+	if (!(typeof room.t !== 'undefined' && room.t === 'l' && room.v && room.v.token)) {
+		return message;
+	}
+	const sentByAgent = !message.token;
+	if (sentByAgent) {
+		const department = LivechatDepartment.findOneById(room.departmentId);
```

Comment:
Please, check `room.departmentId` before finding the department, then we can avoid unnecessary mongo process.